src/dc-factory.ts

```diff
@@ -0,0 +1,164 @@
+import { findElementsForInit, isElementWithinLazyParent } from './dc-dom';
+import { DcBaseComponent } from 'src/dc-base-component'
+
+/**
+ * @typedef {string} ComponentState
+ * */
+
+/**
+ * @type {ComponentState}
+ */
+const STATE_NOT_INITED = 'not-inited';
+/**
+ * @type {ComponentState}
+ */
+const STATE_INITIALIZING = 'initializing';
+/**
+ * @type {ComponentState}
+ */
+const STATE_LAZY_WAITING = 'lazy-waiting';
+/**
+ * @type {ComponentState}
+ */
+const STATE_CREATED = 'created';
+/**
+ * @type {ComponentState}
+ */
+const STATE_ERROR = 'error';
+
+interface StatedComponent {
+    element: HTMLElement;
+    instance: DcBaseComponent | undefined
+    state: string
+}
+
+interface RegisteredComponent {
+    _Class: typeof DcBaseComponent;
+    selector: string | Function | undefined;
+    components: StatedComponent[]
+}
+
+class DcFactory {
+    private _registeredComponents:  RegisteredComponent[] = [];
+
+    /**
+     *
+     * @param {typeof DcBaseComponent} ComponentClass
+     * @param {Function|string} selector that indicates how we should search that component elements
+     */
+    public register(ComponentClass: typeof DcBaseComponent, selector?: string | Function | undefined): void {
+        console.log(selector);
```

Comment:
Should we get rid off of the console statement here?

Reply:
oh yes, I forgot to remove it after debugging